pkg/autodiscovery/listeners/kube_services.go

```diff
@@ -69,10 +71,13 @@ func NewKubeServiceListener() (ServiceListener, error) {
 		return nil, fmt.Errorf("cannot get service informer: %s", err)
 	}
 
+	_, kubeServicesFileEnabled := conf.EnabledProviders["kube_services_file"]
```

Comment:
nit: Technically speaking, the listeners should not need to know how to check this, having an interface providing `IsProviderEnabled` is an option.
Also as this is a fixed list, it could even be a package function initialized when AD is setup by `cmd/agent/common`.

And as we have a list of `const`, might as well use the `const` here.

Reply:
This makes perfect sense, thanks! addressed in https://github.com/DataDog/datadog-agent/pull/10073/commits/d680f1665b5ade73be5925a087ab818afb914674

Reply:
I chose the the interface option instead of the function package to avoid another potential singleton initialized by AD.